pyam/netcdf.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+import datetime as dt
+import pandas as pd
+
+try:
+    import xarray as xr
+except ImportError:
+    print(
+        "Xarray is required, to install run 'pip.main(['install', '--user', 'xarray'])' "
+    )
+from pyam.core import IamDataFrame
+from pyam.utils import META_IDX, IAMC_IDX
+
+
+def read_netcdf(path):
+    """Read timeseries or year-based data and meta-indicators from netCDF
+
+    Parameters
+    ----------
+    path : :class:`pathlib.Path` or file-like object
+        Scenario data file in netCDF format.
+
+    Returns
+    ----------
+    IamDataFrame with meta indicators if available
+
+    """
+
+    _ds = xr.open_dataset(path)
+    NETCDF_IDX = ["time", "model", "scenario", "region"]
+    _list_variables = [i for i in _ds.to_dict()["data_vars"].keys()]
+    _meta = []
```

Comment:
Suggest to introduce `_meta` at the same place as `dfs`.

Reply:
I still think that `_data` and `_meta` should be defined at the same place in the script.